### PR TITLE
Add usage metrics to feature flags list

### PR DESCRIPTION
**Pull request by orangejenny**
Summit followup: make it quicker to see how heavily individual feature flags are used, to help identify good candidates for productizing/killing.

This is quick and dirty (read: not performant).

screen shot 2015-10-08 at 3 58 39 pm

code buddy @nickpell 
fyi @dimagi/product

### CODE REVIEW

File: corehq/apps/toggle_ui/views.py [hunk 1]
```diff
@@ -31,9 +32,28 @@ def page_url(self):
 
     @property
     def page_context(self):
+        toggles = list(all_toggles())
```
Comment (benrudolph) [inline corehq/apps/toggle_ui/views.py]: if this returns a generator you don't have to convert it a list since you're iterating it over it once

**Comment by krsdimagi** [comment]
nice!

File: corehq/apps/toggle_ui/views.py (hunk 1 above)
Comment (orangejenny) [inline corehq/apps/toggle_ui/views.py]: the template iterates over it, too

**Comment by benrudolph** [comment]
this is sweet

**Comment by amsagoff** [comment]
one idea for performance: these will be mostly irrelevant unless we
proactively decide to do an analysis. So we don't have to run this in
real-time or even that regularly - we could have an "update" button trigger
a recalculation.

File: corehq/apps/toggle_ui/views.py [hunk 2]
```diff
@@ -31,9 +32,28 @@ def page_url(self):
...
+        domain_counts = {}
+        user_counts = {}
+        for t in toggles:
+            counter = Counter()
+            try:
+                usage = Toggle.get(t.slug)
+            except ResourceNotFound:
+                domain_counts[t.slug] = 0
+                user_counts[t.slug] = 0
+                continue
```
Comment (czue) [inline corehq/apps/toggle_ui/views.py]: nitpick, but using a `try`/`else` construct would avoid needing a continue and make the control flow a little clearer

**Comment by czue** [comment]
a simpler way to manage the performance problem would just be to have a button (or even a secret url param since it's internal) that adds those columns if set

**Comment by czue** [comment]
(this is great btw!)

**Comment by orangejenny** [comment]
> a simpler way to manage the performance problem would just be to have a button (or even a secret url param since it's internal) that adds those columns if set

yeah, that's simpler - will do

**Comment by millerdev** [comment]
:heart:

**Comment by orangejenny** [comment]
screen shot 2015-10-09 at 10 16 05 am

**Comment by benrudolph** [comment]
nice, LGTM after tests, can't wait to get this on prod

File: corehq/apps/toggle_ui/views.py [hunk 3]
```diff
@@ -30,10 +31,36 @@ def page_url(self):
         return reverse(self.urlname)
 
     @property
+    def show_usage(self):
+        return self.request.GET.get('show_usage') == 'true'
```
Comment (benrudolph) [inline corehq/apps/toggle_ui/views.py]: will this crash hard if `show_usage` doesn't exist?

Comment (orangejenny) [inline corehq/apps/toggle_ui/views.py]: doesn't seem to

File: corehq/apps/toggle_ui/views.py [hunk 4]
```diff
@@ -30,10 +31,36 @@ def page_url(self):
         return reverse(self.urlname)
 
     @property
+    def show_usage(self):
```
Comment (millerdev) [inline corehq/apps/toggle_ui/views.py]: Nitpick: `show_usage` seems to be an action (like a function that would be called to do something). Consider renaming to `should_show_usage`.

**Comment by snopoke** [comment]
Nice! Worth adding a notice or something for random toggles?

**Comment by orangejenny** [comment]
@snopoke added icon
screen shot 2015-10-09 at 12 33 56 pm

File: corehq/apps/toggle_ui/views.py (hunk 4 above)
Comment (orangejenny) [inline corehq/apps/toggle_ui/views.py]: mmm...I see your point, but I'm going to say the action is "show" and leave it as is :-p

**Comment by snopoke** [comment]
Awesome

**Comment by benrudolph** [comment]
gonna go ahead and merge, but agree would be interesting to see percentage for random toggles

**Comment by orangejenny** [comment]
Sure, randomness percentage could be useful to devs. Feel free to add it.

**Pull request merged**